server/data/data-library/src/main/java/org/apache/james/domainlist/lib/AbstractDomainList.java

```diff
@@ -69,8 +69,12 @@ public void configure(HierarchicalConfiguration config) throws ConfigurationExce
     }
 
     @Override
-    public String getDefaultDomain() {
-        return defaultDomain;
+    public String getDefaultDomain() throws DomainListException {
```

Comment:
There's still a problem, you are replacing the NPE by another Exception; but the init will still fail, no?